haystack/components/converters/docx.py

```diff
@@ -79,11 +82,16 @@ class DOCXToDocument:
     ```
     """
 
-    def __init__(self):
+    def __init__(self, table_format: Literal["markdown", "csv"] = "markdown"):
         """
         Create a DOCXToDocument component.
+
+        :param table_format: The format for table output. Can be either "markdown" or "csv". Defaults to "markdown".
         """
         docx_import.check()
+        if table_format not in ["markdown", "csv"]:
```

Comment:
I'd rather we just exported tables as a CSV dataframe and leave the markdown conversion for a separate component. That way we don't tie the rendering format to the component that extracts the data.

Reply:
That's cool with me, let's ask internal if that's ok with them - @sjrl ?
Another option is to have Callable for table conversion and by default we use csv. Or if not callable a PyPDFConverter type of Protocol, wdyt?

Reply:
Hmm @shadeMe the only issue foresee with this is that I don't believe we are extracting the tables separately into their own Documents (by request). 

We have found that keeping the table with the surrounding text in a single Document has improved performance since the table can usually be better retrieved due to contextual queues from the surrounding text. However, since we do this it would not then be easy to modify the CSV into a different format later so I do see a reason to support different formatting options in this component. 

As a side note being able to optionally export the tables separately into their own Documents would also be helpful to allow for more table specific processing as @shadeMe suggests.

Reply:
Ah, fair point. In that case, @vblagoje let's stick to your original proposal for now (+ the enum).

Reply:
Ok thanks guys, will make adjustment as agreed. Markdown+csv internal conversion methods + enum it is.